overlord.go

```diff
@@ -220,65 +219,85 @@ func iterate() {
 	// log.Println("Start iteration")
 
 	resources := make(map[Lookable]map[*Resource]bool)
-	state     := make(map[string]map[string]bool)
+	state := make(map[string]map[string]bool)
 
 	//load resources definition files
 	resourcesDir, err := os.Open(resourcesDirName)
-	defer func(){resourcesDir.Close()}()
-	if err != nil { log.Fatal(err) }
+	defer func() { resourcesDir.Close() }()
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	resourcesFiles, err := resourcesDir.Readdir(0)
-	if err != nil { log.Fatal(err) }
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	for _, resourceFile := range resourcesFiles {
-		
-		if filepath.Ext(resourceFile.Name()) != ".toml" ||  resourceFile.IsDir() {continue}
-		
+
+		if filepath.Ext(resourceFile.Name()) != ".toml" || resourceFile.IsDir() {
+			continue
+		}
+
 		var rc *ResourceConfig
 		_, err := toml.DecodeFile(filepath.Join(resourcesDirName, resourceFile.Name()), &rc)
-		if err != nil { log.Fatal(err) }
+		if err != nil {
+			log.Fatal(err)
+		}
 
 		// log.Println("Read File", resourceFile.Name(), ":", rc)
 
 		for _, group := range rc.Resource.Groups {
-			if resources[group] == nil {resources[group]=make(map[*Resource]bool)}
+			if resources[group] == nil {
+				resources[group] = make(map[*Resource]bool)
+			}
 			resources[group][&rc.Resource] = true
 		}
 
 		for _, tag := range rc.Resource.Tags {
-			if resources[tag] == nil {resources[tag]=make(map[*Resource]bool)}
+			if resources[tag] == nil {
+				resources[tag] = make(map[*Resource]bool)
+			}
 			resources[tag][&rc.Resource] = true
 		}
 	}
 
 	//load state file
 	err = os.MkdirAll(filepath.Dir(stateFileName), 0777)
-	if err != nil { log.Fatal(err) }
+	if err != nil {
+		log.Fatal(err)
+	}
 
 	_, err = toml.DecodeFile(stateFileName, &state)
-	if err != nil && !os.IsNotExist(err) { log.Fatal(err) }	
+	if err != nil && !os.IsNotExist(err) {
+		log.Fatal(err)
+	}
 
 	// log.Println("Load state from", stateFileName, ":", state)
 
-
-
 	// log.Println("Find Resources to update")
 
 	resourcesToUpdate := make(map[*Resource]bool)
 	newState := make(map[string]map[string]bool)
-	
+
 	//find group ips to update
 	for g, resourcesset := range resources {
-		
+
 		//substitute group name by var env if existing
 
 		group := g.String()
-		ips := g.LookupIPs()
+		ips, err := g.LookupIPs()
+
+		// if some AWS API calls failed during the IPs lookup, stop here and exit
+		// it will keep the dest file unmodified and won't execute the reload command.
+		if err != nil {
+			log.Fatal(err)
```

Comment:
The application may be restarted as soon as it leave and so enter a kind of infinite loop crash -> restart that will consume CPU.

Reply:
The upstart script wait 3 seconds before restarting overlord, so the impact on the CPU will be pretty low.

Reply:
I managed to test a failure case by disabling the eth0 interface. I got the following error:
`2019/06/11 14:41:13 RequestError: send request failed
caused by: Post https://ec2.eu-west-1.amazonaws.com/: read tcp 10.188.4.33:42164->54.239.39.130:443: read: connection reset by peer` and overlord has been stopped. 
So it is working like requested in the code.

Another option would be to only exit the _iterate_ function rather than exiting the whole process. What do you think @gaetanmorice?

Yet another solution would be to implement kind of exponential backoff in `post-stop exec sleep x`

Reply:
Ok for the test.

I think you are right we can rely on the wait on the upstart script